kunta-api-server/src/main/java/fi/otavanopisto/kuntaapi/server/integrations/mikkelinyt/MikkeliNytEventProvider.java

```diff
@@ -0,0 +1,360 @@
+    String baseUrl = organizationSettingController.getSettingValue(organizationId, MikkeliNytConsts.ORGANIZATION_SETTING_BASEURL);
+    
+    Map<String, Object> queryParams = new HashMap<>();
+    queryParams.put("apiKey", apiKey);
+    
+    if (StringUtils.isNotBlank(location)) {
+       queryParams.put("location", location);
+    } else {
+      logger.warning("location not specified. Returning unfiltered event list");
+    }
+    
+    return httpClient.doGETRequest(baseUrl, "/json.php", new GenericHttpClient.ResultType<fi.otavanopisto.mikkelinyt.model.EventsResponse>() {}, queryParams);
+  }
+
+  private AttachmentData getImageData(OrganizationId organizationId, AttachmentId imageId) {
+    AttachmentId mikkeliNytId = idController.translateAttachmentId(imageId, MikkeliNytConsts.IDENTIFIER_NAME);
+    if (mikkeliNytId == null) {
+      logger.severe(String.format("Failed to translate %s into MikkeliNyt id", imageId.toString()));
+      return null;
+    }
+    
+    String imageBaseUrl = organizationSettingController.getSettingValue(organizationId, MikkeliNytConsts.ORGANIZATION_SETTING_IMAGEBASEURL);
+    if (StringUtils.isNotBlank(imageBaseUrl)) {
+      String imageUrl = String.format("%s%s", imageBaseUrl, mikkeliNytId.getId());
+      Response<AttachmentData> imageDataResponse = getImageData(imageUrl);
+      if (imageDataResponse.isOk()) {
+        return imageDataResponse.getResponseEntity();
+      } else {
+        logger.severe(String.format("Request to find image (%s) data failed on [%d] %s", imageId.toString(), imageDataResponse.getStatus(), imageDataResponse.getMessage()));
+      }
+    }
+    
+    logger.severe(String.format("Image imageBaseUrl has not been configured properly for organization %s", organizationId));
+    
+    return null;
+  }
+  
+  private Response<AttachmentData> getImageData(String imageUrl) {
+    try {
+      CloseableHttpClient imageClient = HttpClients.createDefault();
+      try {
+        HttpGet httpGet = new HttpGet(imageUrl);
+        
+        CloseableHttpResponse response = imageClient.execute(httpGet);
+        try {
+          StatusLine statusLine = response.getStatusLine();
+          int statusCode = statusLine.getStatusCode();
+          String message = statusLine.getReasonPhrase();
+          InputStream data = IOUtils.toBufferedInputStream(response.getEntity().getContent());
+          String type = response.getEntity().getContentType().getValue();
+          return new Response<>(statusCode, message, new AttachmentData(type, data));
+        } finally {
+          response.close();
+        }
+      } finally {
+        imageClient.close();
+      }
+    } catch (IOException e) {
+      logger.log(Level.WARNING, String.format("Failed to fetch event imagae from %s", imageUrl), e);
+      return new Response<>(500, "Interval Server Error", null);
+    }
+  }
+  
+  private AttachmentId getImageAttachmentId(String url) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "getImageAttachmentId" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)